teamsupport/models.py

```diff
@@ -0,0 +1,66 @@
+from property_caching import cached_property
+
+
+class Ticket(object):
+    _data = None
+
+    def __init__(self, client, ticket_id=None, data=None):
+        self.client = client
+        if data is not None:
+            self._data = data
+            self.id = data.find('TicketID').text
+        elif ticket_id:
+            self.id = ticket_id
+        elif not (ticket_id or data):
+            raise TypeError(
+                "__init__() needs either a 'ticket_id' or 'data' argument "
+                "(neither given)")
```

Comment:
I used `TypeError` because it's what Python raises if you give a function 1 argument when it needs 2, but `ValueError` makes more sense.

```
In [1]: class A(object):
   ...:     def __init__(self, b):
   ...:         self.b = b
   ...:

In [2]: A()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-3cd318a12eea> in <module>()
----> 1 A()
TypeError: __init__() takes exactly 2 arguments (1 given)
```